src/reviewer-protocol/ThriveReview.sol

```diff
@@ -0,0 +1,616 @@
+        for (uint256 i = 0; i < reviewIds_.length; i++) {
+            deletePendingReview(reviewIds_[i]);
+        }
+    }
+
+
+    // @inheritdoc IThriveReview
+    function deletePendingReview(uint256 reviewId_) public {
+
+        // Require that the review is in the "COMMITED" status
+        require(reviews[reviewId_].status == ReviewStatus.COMMITED, "Review is not in 'COMMITED' status");
+        
+        // Require that the reviews' deadline has passed
+        require(block.timestamp > reviews[reviewId_].deadline, "Review deadline has not passed");
+
+        // Fetch the review data
+        uint256 submissionId = reviews[reviewId_].submissionId;
+        address reviewer = reviews[reviewId_].reviewer;
+
+        // User un-committed from review
+        userCommitedToReview[reviewer][submissionId] = false;
+
+        // Decrement the counter of committed reviews
+        committedReviewsPerSubmissionCounter[submissionId]--;
+
+        // Delete the review from `reviews` mapping
+        delete reviews[reviewId_];
+
+
+        ////// EVENT
+
+    }    
+
+    // @inheritdoc IThriveReview
+    function reachDecisionOnSubmission(uint256 submissionId_) external 
+        submissionPending(submissionId_)
+    {
+        _reachDecisionOnSubmission(submissionId_);
+    }
+
+
+    // @inheritdoc IThriveReview
+    function _reachDecisionOnSubmission(uint256 submissionId_) internal {
+        
+        // Fetch the submission from storage and copy to memory
+        Submission storage submission = submissions[submissionId_];
+
+        // We can put this "if statement" below as a require if this function is not called automatically on createReview function.
+
+        // Check if the submission has enough reviews to reach a decision
+        if (submission.reviewCount >= reviewConfiguration.minimumReviews) {
+
+            // Calculate the ratio of accepted reviews
+            uint256 acceptedReviews = submission.acceptedReviewsCount;
+            uint256 acceptedRatio = (acceptedReviews * SCALER) / submission.reviewCount;
+
+            // Calculate the ratio of rejected reviews
+            uint256 rejectedReviews = submission.rejectedReviewsCount;
+            uint256 rejectedRatio = (rejectedReviews * SCALER) / submission.reviewCount;
+
+            // Check if the ratio is above the agreement threshold
+            if (acceptedRatio >= reviewConfiguration.agreementThreshold) {
+
+                // Confirm the submission is no longer PENDING
+                submission.status = SubmissionStatus.FINALIZED;
+
+                // Submission is ACCEPTED
+                submission.decision = Decision.ACCEPTED;
+
+                // If work unit contract is set, confirm the submission on the work unit contract
+                if (hasWorkerUnitContract()) {
+                    IThriveWorkerUnit(workerUnitAddress).confirm(submission.contributor, submission.submissionMetadata);
```

Comment:
this sends funds to contributor and reward to ThriveReview, were do we distribute funds to reviewers?